src/scripts/setup_external_storage/setup_external_storage.py

```diff
@@ -0,0 +1,61 @@
+"""
+Create an STS-enabled folder on Synapse over an S3 location.
+"""
+
+import os
+import json
+import argparse
+
+import synapseclient
+
+
+def read_args():
+    parser = argparse.ArgumentParser(
+        description="Create an STS-enabled folder on Synapse over an S3 location."
+    )
+    parser.add_argument(
+        "--synapse-parent", help="Synapse ID of the parent folder/project"
+    )
+    parser.add_argument(
+        "--synapse-folder-name",
+        help=(
+            "Name of the Synapse folder to use with the external " "storage location."
+        ),
+    )
+    parser.add_argument("--s3-bucket", help="S3 bucket name")
+    parser.add_argument(
+        "--s3-key",
+        default=None,
+        help="S3 key to serve as the root. `/` (root) by default.",
+    )
+    parser.add_argument(
+        "--sts-enabled",
+        action="store_true",
+        help="Whether this storage location should be STS enabled",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = read_args()
+    syn = synapseclient.login()
+    storage_location = syn.create_s3_storage_location(
+        parent=args.synapse_parent,
+        folder_name=args.synapse_folder_name,
+        bucket_name=args.s3_bucket,
+        base_key=args.s3_key,
+        sts_enabled=args.sts_enabled,
+    )
+    storage_location_info = {
+        k: v
+        for k, v in zip(
+            ["synapse_folder", "storage_location", "synapse_project"], storage_location
+        )
+    }
```

Comment:
I like the neat trick here that you do to parse through the tuple `create_s3_storage_location` returns.  The reason I like my suggested correction is it a quicker way to knowing that `create_s3_storage_location` returns a tuple. Thoughts? 

```suggestion
    synapse_folder, storage_location, synapse_project = syn.create_s3_storage_location(
        parent=args.synapse_parent,
        folder_name=args.synapse_folder_name,
        bucket_name=args.s3_bucket,
        base_key=args.s3_key,
        sts_enabled=args.sts_enabled,
    )
    storage_location_info = {
        "synapse_folder": synapse_folder,
        "storage_location": storage_location,
        "synapse_project": synapse_project
    }
```

Reply:
Is the preference to just get each object of the tuple `storage_location` by index then like this? 
```
storage_location_info = {
        "synapse_folder": storage_location[0],
        "storage_location": storage_location[1],
        "synapse_project": storage_location[2]
    }
```

Reply:
Yes I agree, the code seems more intuitive as well.